**step-1**
**Cleaning**

In [ ]:
import pandas as pd

# Load dataset
df = pd.read_csv("Placement_Data_Full_Class.csv")

# Check first 5 rows
df.head()


**step-2 checking missing values**
    



In [33]:
df.isna().sum()

sl_no              0
gender             0
ssc_p              0
ssc_b              0
hsc_p              0
hsc_b              0
hsc_s              0
degree_p           0
degree_t           0
workex             0
etest_p            0
specialisation     0
mba_p              0
status             0
salary            67
dtype: int64

In [5]:

df.duplicated().sum()


0

In [7]:
df.dtypes


sl_no               int64
gender             object
ssc_p             float64
ssc_b              object
hsc_p             float64
hsc_b              object
hsc_s              object
degree_p          float64
degree_t           object
workex             object
etest_p           float64
specialisation     object
mba_p             float64
status             object
salary            float64
dtype: object

**step-4**
**EDA — Placed vs Not Placed**

**Business Question 1**

In [11]:
# Overall placement count
df['status'].value_counts()


status
Placed        148
Not Placed     67
Name: count, dtype: int64

In [13]:
#Placement percentage:
placement_rate = df['status'].value_counts(normalize=True) * 100
placement_rate


status
Placed        68.837209
Not Placed    31.162791
Name: proportion, dtype: float64

**Business Question 2**

In [20]:
#Gender-wise placement rate
pd.crosstab(df['gender'], df['status'], normalize='index') * 100


status  Not Placed     Placed
gender                       
F        36.842105  63.157895
M        28.057554  71.942446

**Skill Gap – Academics vs Placement**

In [39]:
df.groupby('status')[['ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p']].mean()


ssc_p      hsc_p   degree_p    etest_p      mba_p
status                                                           
Not Placed  57.544030  58.395522  61.134179  69.587910  61.612836
Placed      71.721486  69.926554  68.740541  73.238041  62.579392

In [41]:
pd.crosstab(df['workex'], df['status'], normalize='index') * 100


status  Not Placed     Placed
workex                       
No       40.425532  59.574468
Yes      13.513514  86.486486

**Salary Analysis (Placed students)**

In [46]:
# Salary stats (only placed students)
df[df['status'] == 'Placed']['salary'].describe()


count       148.000000
mean     288655.405405
std       93457.452420
min      200000.000000
25%      240000.000000
50%      265000.000000
75%      300000.000000
max      940000.000000
Name: salary, dtype: float64

In [48]:
#Salary by Specialisation:
df[df['status'] == 'Placed'].groupby('specialisation')['salary'].mean()


specialisation
Mkt&Fin    298852.631579
Mkt&HR     270377.358491
Name: salary, dtype: float64

**Final Insights:**

**1. Overall placement rate is ~69%.**

**2. Male students have slightly higher placement rate than females.**
    
**3. Higher academic scores increase chances of placement.**
    
**4. Work experience greatly improves placement probability.**

**5. Average salary is around 2.8 LPA (max up to 9.4 LPA).**

**6. Mkt&Fin students earn more on average than Mkt&HR.**